.github/workflows/msvc.yml

```diff
@@ -0,0 +1,14 @@
+name: Visual Studio build
+on: [push, pull_request]
```

Comment:
NOTE: this way there will be two instances of the build started when a PR created: one for PR and the another one for a push to branch.
See [here](https://github.com/orgs/community/discussions/26276). Right now it is not triggered only because this rule is not in master yet. That discussion provides a good solution to that problem (the one marked with "answer").

Reply:
This is good. Please help to follow the answer and push the fix. Thanks,

Reply:
```suggestion
on:
  push:
    branches:
    - master
  pull_request:
    branches:
    - master
```

Reply:
when other branches are needed - those lists should be extended